test/uk/gov/hmrc/tai/controllers/income/IncomeControllerSpec.scala

```diff
@@ -147,61 +158,442 @@ class IncomeControllerSpec extends PlaySpec
+
+    "return list of live and matched Employments & TaxCodeIncomes as IncomeSource for a given year" in {
+      when(mockIncomeService.taxCodeIncomes(any(), Matchers.eq(TaxYear().next))(any()))
+        .thenReturn(Future.successful(taxCodeIncomes))
+
+      when(mockEmploymentService.employments(Matchers.eq(nino), Matchers.eq(TaxYear().next))(any[HeaderCarrier]))
+        .thenReturn(Future.successful(employments))
+
+      when(mockIncomeService.employments(any(), Matchers.eq(nino), Matchers.eq(TaxYear().next))(any()))
+        .thenReturn(Future.successful(employments))
+
+      val sut = createSUT(incomeService = mockIncomeService, employmentService = mockEmploymentService, authentication = loggedInAuthenticationPredicate)
+      val result = sut.matchedTaxCodeIncomesForYear(nino, TaxYear().next, EmploymentIncome, Live)(FakeRequest())
+
+      val expectedJson = Json.obj(
+        "data" -> Json.arr(Json.toJson(IncomeSource(taxCodeIncomes(1), employment))),
+        "links" -> Json.arr()
+      )
+
+      contentAsJson(result) mustBe expectedJson
+    }
+
+    "return list of ceased and matched Employments & TaxCodeIncomes as IncomeSource for a given year" in {
+
+      when(mockIncomeService.taxCodeIncomes(any(), Matchers.eq(TaxYear().next))(any()))
+        .thenReturn(Future.successful(Seq(
+          TaxCodeIncome(EmploymentIncome, Some(2), BigDecimal(0),
+            EmploymentIncome.toString, "1100L", "Employer2", OtherBasisOperation, Ceased, BigDecimal(321.12), BigDecimal(0), BigDecimal(0))
+        )))
+
+      when(mockEmploymentService.employments(Matchers.eq(nino), Matchers.eq(TaxYear().next))(any[HeaderCarrier]))
+        .thenReturn(Future.successful(employments))
+
+      when(mockIncomeService.employments(any(), Matchers.eq(nino), Matchers.eq(TaxYear().next))(any()))
+        .thenReturn(Future.successful(employments))
+
+      val sut = createSUT(incomeService = mockIncomeService, employmentService = mockEmploymentService, authentication = loggedInAuthenticationPredicate)
+      val result = sut.matchedTaxCodeIncomesForYear(nino, TaxYear().next, EmploymentIncome, Ceased)(FakeRequest())
+
+      val expectedJson = Json.obj(
+        "data" -> Json.arr(Json.toJson(IncomeSource(taxCodeIncomes(1).copy(status = Ceased), employment))),
+        "links" -> Json.arr()
+      )
+
+      contentAsJson(result) mustBe expectedJson
+    }
+
+    "return list of potentially ceased and matched Employments & TaxCodeIncomes as IncomeSource for a given year" in {
+
+      when(mockIncomeService.taxCodeIncomes(any(), Matchers.eq(TaxYear().next))(any()))
+        .thenReturn(Future.successful(Seq(
+          TaxCodeIncome(EmploymentIncome, Some(2), BigDecimal(0),
+            EmploymentIncome.toString, "1100L", "Employer2", OtherBasisOperation, PotentiallyCeased, BigDecimal(321.12), BigDecimal(0), BigDecimal(0))
+        )))
+
+      when(mockEmploymentService.employments(Matchers.eq(nino), Matchers.eq(TaxYear().next))(any[HeaderCarrier]))
```

Comment:
employment service is not called during this test so you don't need to mock this call out